ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -636,14 +639,44 @@ private void handleAdminRequest(Request request) throws InterruptedException, IO
               error = ServerErrorCode.Bad_Request;
             }
           }
+          response = new AdminResponse(adminRequest.getCorrelationId(), adminRequest.getClientId(), error);
+          break;
+        case CatchupStatus:
+          error = ServerErrorCode.No_Error;
```

Comment:
This function is getting pretty long. Is there any way to break it up into handler functions for each admin req type?

Reply:
Did the best I could. Didn't move out the metrics objects setting - let me know if you think it can be improved further (don't want to create new classes for this).

Reply:
Thanks, this looks better!